Plan/nukkit/src/main/java/com/djrapitops/plan/addons/placeholderapi/NukkitPlaceholderRegistrar.java

```diff
@@ -56,27 +60,36 @@ public NukkitPlaceholderRegistrar(
 
     public void register() {
         PlaceholderAPI api = PlaceholderAPI.getInstance();
-        placeholders.getPlaceholders().forEach((name, loader) ->
-                api.visitorSensitivePlaceholder(name, (player, params) -> {
-                            try {
-                                return loader.apply(getPlayer(player), params.get());
-                            } catch (Exception e) {
-                                errorLogger.warn(e, ErrorContext.builder().related("Registering PlaceholderAPI").build());
-                                return null;
-                            }
-                        }
-                ));
+        placeholders.getPlaceholders().forEach((name, loader) -> api.builder(name, Serializable.class)
+                .visitorLoader(entry -> {
+                    try {
+                        return loader.apply(
+                                getPlayer(entry.getPlayer()),
+                                entry.getParameters().getAll().stream()
+                                        .map(Parameter::getValue)
+                                        .collect(Collectors.toList())
+                        );
+                    } catch (Exception e) {
+                        errorLogger.warn(e, ErrorContext.builder().related("Registering PlaceholderAPI").build());
+                        return null;
+                    }
+                }).build()
+        );
 
-        placeholders.getStaticPlaceholders().forEach((name, loader) ->
-                api.staticPlaceholder(name, params -> {
-                            try {
-                                return loader.apply(params.get());
-                            } catch (Exception e) {
-                                errorLogger.warn(e, ErrorContext.builder().related("Registering PlaceholderAPI").build());
-                                return null;
-                            }
-                        }
-                ));
+        placeholders.getStaticPlaceholders().forEach((name, loader) -> api.builder(name, Serializable.class)
+                .loader(entry -> {
```

Comment:
What is entry - is loader looping through a Map of some sort? (`entry` usually refers to `Map.Entry`). Consider renaming this variable

Reply:
I often see `entry` reffering to some kind of data class in general. Which this is the case, where entry holds player instance, parameters and placeholder context. Otherwise having many lambda parameters would make it less readable in my opinion. And also it is used as lambda receiver in kotlin. So should I rename it to placeholderEntry? Or do you have any other idea?

Reply:
I guess `options` might make sense since parameters is kinda already taken by getParameters()